File: mobile/lib/domain/repositories/patient_repository.dart
import 'package:patient_management_app/domain/entities/patient_entity.dart';
import 'package:patient_management_app/domain/entities/appointment_entity.dart';
import 'package:patient_management_app/domain/entities/medical_record_entity.dart';

abstract class PatientRepository {
  Future<List<PatientEntity>> getPatients({String? search, int page = 1, int pageSize = 20});
  Future<PatientEntity> getPatientById(String id);
  Future<PatientEntity> createPatient(PatientEntity patient);
  Future<PatientEntity> updatePatient(PatientEntity patient);
  Future<bool> deletePatient(String id);
  Future<List<AppointmentEntity>> getPatientAppointments(String id);
  Future<List<MedicalRecordEntity>> getPatientMedicalRecords(String id);
  Future<PatientEntity> getMyProfile();
}

File: web/src/data/repositories/PatientRepository.ts
import api from '@/core/api';
import { IPatientRepository } from '@/domain/repositories';
import { Patient } from '@/domain/entities';

export class PatientRepository implements IPatientRepository {
  async getPatients(search?: string, page = 1, pageSize = 20) {
    const response = await api.get('/Patients', {
      params: { search, page, pageSize }
    });
    return {
      data: response.data.data,
      totalCount: response.data.pagination.totalCount
    };
  }

  async getPatientById(id: string) {
    const response = await api.get(`/Patients/${id}`);
    return response.data.data;
  }

  async createPatient(patient: Partial<Patient>) {
    const response = await api.post('/Patients', patient);
    return response.data.data;
  }

  async updatePatient(id: string, patient: Partial<Patient>) {
    const response = await api.put(`/Patients/${id}`, patient);
    return response.data.data;
  }

  async deletePatient(id: string) {
    await api.delete(`/Patients/${id}`);
  }
}
